src/app/destiny2/d2-definitions.ts

```diff
@@ -179,6 +179,9 @@ export function buildDefinitionsFromManifest(db: AllDestinyManifestComponents) {
         if (!dbTable) {
           throw new Error(`Table ${table} does not exist in the manifest`);
         }
+        if (id < 1 || id > Number.MAX_SAFE_INTEGER) {
```

Comment:
Number.isSafeInteger() exists for this, and also ensures no decimal place

Reply:
(to replace the right half of the check)

Reply:
I think I did what you proposed

Reply:
:+1: